File: test-phase0.js
const fs = require('fs');

// Read and parse the rawWords.json file
const rawWords = JSON.parse(fs.readFileSync('rawWords.json', 'utf8'));

// Test 1: Check JSON structure
console.log('Test 1: JSON Structure');
console.log('---------------------');
console.log(`Total words: ${rawWords.length}`);
console.log(`All objects have 'word' property: ${rawWords.every(w => 'word' in w)}`);
console.log(`All objects have 'type' property: ${rawWords.every(w => 'type' in w)}`);
console.log(`All types are arrays: ${rawWords.every(w => Array.isArray(w.type))}`);
console.log(`All types have at least one value: ${rawWords.every(w => w.type.length > 0)}`);

// Test 2: Check for duplicates
console.log('\nTest 2: Duplicate Check');
console.log('---------------------');
const words = rawWords.map(w => w.word);
const uniqueWords = new Set(words);
console.log(`Duplicate words found: ${words.length !== uniqueWords.size}`);
if (words.length !== uniqueWords.size) {
    const duplicates = words.filter((word, index) => words.indexOf(word) !== index);
    console.log('Duplicate words:', duplicates);
}

// Test 3: Check type categories
console.log('\nTest 3: Type Categories');
console.log('---------------------');
const allTypes = new Set(rawWords.flatMap(w => w.type));
console.log('All unique types:', Array.from(allTypes).sort());

// Test 4: Compare with original wordList
console.log('\nTest 4: Compare with original wordList');
console.log('---------------------');
const originalWords = [
    "Cat", "Dog", "Ball", "Car", "Ford", "Alien", "Bug", "Bee", "Jaguar", "Kia",
    "Deer", "Cow", "Bat", "Hat", "Pig", "Truck", "Person", "Zebra", "Lion", "Sloth",
    "Panda", "Frog", "Clam", "Box", "Book", "Lexus", "Honda", "Pacifica", "Boy", "Girl", "BMW"
];
const rawWordsList = rawWords.map(w => w.word);
const missingWords = originalWords.filter(w => !rawWordsList.includes(w));
const extraWords = rawWordsList.filter(w => !originalWords.includes(w));

console.log(`Missing words: ${missingWords.length ? missingWords : 'None'}`);
console.log(`Extra words: ${extraWords.length ? extraWords : 'None'}`); 

File: test-phase0.py
import json

# Read and parse the rawWords.json file
with open('rawWords.json', 'r') as f:
    rawWords = json.load(f)

# Test 1: Check JSON structure
print('Test 1: JSON Structure')
print('---------------------')
print(f'Total words: {len(rawWords)}')
print(f"All objects have 'word' property: {all('word' in w for w in rawWords)}")
print(f"All objects have 'type' property: {all('type' in w for w in rawWords)}")
print(f"All types are arrays: {all(isinstance(w['type'], list) for w in rawWords)}")
print(f"All types have at least one value: {all(len(w['type']) > 0 for w in rawWords)}")

# Test 2: Check for duplicates
print('\nTest 2: Duplicate Check')
print('---------------------')
words = [w['word'] for w in rawWords]
uniqueWords = set(words)
print(f"Duplicate words found: {len(words) != len(uniqueWords)}")
if len(words) != len(uniqueWords):
    duplicates = [word for word in words if words.count(word) > 1]
    print('Duplicate words:', duplicates)

# Test 3: Check type categories
print('\nTest 3: Type Categories')
print('---------------------')
allTypes = set()
for word in rawWords:
    allTypes.update(word['type'])
print('All unique types:', sorted(list(allTypes)))

# Test 4: Compare with original wordList
print('\nTest 4: Compare with original wordList')
print('---------------------')
originalWords = [
    "Cat", "Dog", "Ball", "Car", "Ford", "Alien", "Bug", "Bee", "Jaguar", "Kia",
    "Deer", "Cow", "Bat", "Hat", "Pig", "Truck", "Person", "Zebra", "Lion", "Sloth",
    "Panda", "Frog", "Clam", "Box", "Book", "Lexus", "Honda", "Pacifica", "Boy", "Girl", "BMW"
]
rawWordsList = [w['word'] for w in rawWords]
missingWords = [w for w in originalWords if w not in rawWordsList]
extraWords = [w for w in rawWordsList if w not in originalWords]

print(f"Missing words: {missingWords if missingWords else 'None'}")
print(f"Extra words: {extraWords if extraWords else 'None'}") 